### PR TITLE
only run one update per frame drawn

### LINKED ISSUE #13965: WASM perf regression: App updates triggered by input events
## Bevy version

* Original: bevy-0.13.2
* Current: bevy-0.14.0-rc.3

## Relevant system information

Please include:

* Rust version: 0.78.0
* Firefox 126.0.1
* Linux/x11
* Nvidia proprietary 550.90.07

## What's performing poorly?

I discovered that after updating my wasm app to 0.14, it feels very sluggish.

Here is a minimal reproducible example:
When idle the square moves in a circle. Moving the cursor or keeping a key pressed, the square will begin to skip around.
```rust
use bevy::{prelude::*, sprite::MaterialMesh2dBundle};

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .insert_resource(A(0))
        .add_systems(Startup, setup)
        .add_systems(Update, run)
        .run();
}

fn setup(
    mut commands: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut materials: ResMut<Assets<ColorMaterial>>,
) {
    commands.spawn(Camera2dBundle::default());
    commands.spawn((
        MaterialMesh2dBundle {
            mesh: meshes.add(Rectangle::default()).into(),
            transform: Transform::default().with_scale(Vec3::splat(128.)),
            material: materials.add(Color::WHITE),
            ..default()
        },
        Angle(0),
    ));
}

#[derive(Debug, Component)]
struct Angle(u8);

#[derive(Resource)]
struct A(u128);

fn run(mut q_angle: Query<(&mut Angle, &mut Transform)>, mut a: ResMut<A>) {
    // do something slow
    for i in 0..10_000_000 {
        a.0 += i * i;
    }

    for (mut angle, mut transform) in &mut q_angle {
        angle.0 = (angle.0 + 1) % 8;
        transform.translation =
            (Vec2::from_angle(angle.0 as f32 / 8.0 * std::f32::consts::TAU) * 250.0).extend(0.0);
    }
}
```

## Additional information

We can see that after 0.14, updates are triggered by pointermove events. Painting only happens about once every 3 frames as a result.  The same happens when keeping a key pressed.

* Before:
image

* After:
image

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Fixes #13965 

## Solution

- Don't run multiple updates for a single frame

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Terrible. Robust though!

**Comment by MScottMcBee** [comment]
A few questions on this:

- Wouldn't this effectively lock all bevy apps to VSync? 
- How does it play with something like bevy_framepace?
- Would it be possible to put it behind some sort of setting?

**Comment by mockersf** [comment]
> * Wouldn't this effectively lock all bevy apps to VSync?

Nop, all the non VSync modes are still driven by the window asking for a redraw, those are just never actually displayed

> * How does it play with something like bevy_framepace?

I don't think there should be any issue, it should actually unbreak it in some case

> * Would it be possible to put it behind some sort of setting?

Sure, but until we're actually able to properly dissociate (and control how it works) the expected behaviour is "only one update per frame"

**Review by MScottMcBee — APPROVED** [review]

**Review by IceSentry — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@mockersf this seems to be causing the examples to fail by never updating on Linux: https://github.com/bevyengine/bevy/actions/runs/9679910502

**Comment by mockersf** [comment]
tried a way that allows more than 1 update before the first redraw, hopefully that will fix linux in CI

File: crates/bevy_winit/src/state.rs [hunk 1]
```diff
@@ -497,9 +503,10 @@ impl<T: Event> ApplicationHandler<T> for WinitAppRunnerState<T> {
         // If the cycle takes more than the wait timeout, it will be re-executed immediately.
         let begin_frame_time = Instant::now();
 
-        if should_update {
+        if self.updates_ran_since_last_redraw <= 0 && should_update {
             // Not redrawing, but the timeout elapsed.
             self.run_app_update();
+            self.updates_ran_since_last_redraw = self.updates_ran_since_last_redraw + 1;
```
Comment (alice-i-cecile) [inline crates/bevy_winit/src/state.rs:509]:
```suggestion
            self.updates_ran_since_last_redraw += 1;
```

**Comment by mockersf** [comment]
after some trials, this should fix the bug and work in CI

**Pull request merged**